src/controllers/users.go

```diff
@@ -0,0 +1,169 @@
+
+	roleBinding, err := u.client.RbacV1().RoleBindings(user.Namespace).Create(u.ctx,
+		prepareRoleBinding(user.Name, user.Role), metav1.CreateOptions{})
+	if err != nil {
+		u.logger.Error(fmt.Sprintf("Could create rolebinding %q with error: %v", user.Name, err.Error()))
+		return userOutput, err
+	}
+	u.logger.Debug(fmt.Sprintf("created roleBinding %q successfully", user.Name))
+	userOutput.Name = roleBinding.Name
+	userOutput.Role = user.Role
+
+	return userOutput, nil
+}
+
+func (u *userController) UpdateUser(user types.UserInput) (types.User, error) {
+	userOutput := types.User{}
+
+	u.logger.Debug(fmt.Sprintf("Trying to update rolebinding %q in %q namespace", user.Name, user.Namespace))
+
+	_, err := u.DeleteUser(types.UserIdentifier{UserName: user.Name, NamespaceName: user.Namespace})
+	if err != nil {
+		return userOutput, err
+	}
+
+	err = retry.OnError(retry.DefaultRetry, func(err error) bool {
+		return err != nil
+
+	}, func() error {
+		_, err := u.AddUser(user)
+		return err
+	})
+
+	if err != nil {
+		u.logger.Error(fmt.Sprintf("Could not update rolebinding %q with error: %v", user.Name, err.Error()))
+		return userOutput, err
+	}
+
+	u.logger.Debug(fmt.Sprintf("updated roleBinding %q successfully", user.Name))
+	userOutput.Name = user.Name
+	userOutput.Role = user.Role
+
+	return userOutput, nil
+}
+
+func (u *userController) DeleteUser(userIdentifier types.UserIdentifier) (types.DeleteUserResponse, error) {
+	u.logger.Debug(fmt.Sprintf("Trying to delete rolebinding %q in %q namespace", userIdentifier.UserName, userIdentifier.NamespaceName))
+
+	if err := u.client.RbacV1().RoleBindings(userIdentifier.NamespaceName).Delete(u.ctx, userIdentifier.UserName, metav1.DeleteOptions{}); err != nil {
+		u.logger.Error(fmt.Sprintf("Could note delete rolebinding %q with error: %v",
+			userIdentifier.UserName, err.Error()))
+		return types.DeleteUserResponse{Message: fmt.Sprintf("Could note delete rolebinding %q with error: %s",
+			userIdentifier.UserName, err.Error())}, err
+	}
+
+	u.logger.Debug(fmt.Sprintf("deleted roleBinding %q successfully", userIdentifier.UserName))
+	return types.DeleteUserResponse{Message: fmt.Sprintf("deleted roleBinding %q successfully", userIdentifier.UserName)}, nil
+}
+
+func prepareRoleBinding(roleBindingName string, role string) *rbacv1.RoleBinding {
+	return &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: roleBindingName,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:     rbacv1.UserKind,
+				Name:     roleBindingName,
+				APIGroup: rbacv1.GroupName,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind:     "ClusterRole",
+			Name:     convertToK8sRoles(role),
+			APIGroup: rbacv1.GroupName,
+		},
+	}
+}
+
+func convertToK8sRoles(requestRole string) string {
+	rolesConvertor := map[string]string{"admin": "capp-user-admin", "contributor": "capp-user-contributor", "viewer": "capp-user-viewer"}
```

Comment:
Please consider it as constants. nit: Also add explain it in the comment why it's needed.